### PR TITLE
New version: Google.JapaneseIME version 2.29.5370.0

### PR DESCRIPTION
**Pull request by kachick**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/144281)

### CODE REVIEW

**Comment by kachick** [comment]
@microsoft-github-policy-service agree

**Comment by kachick** [comment]
How to create this PR
---

> komac --version
komac 2.1.0

```pwsh
komac update --identifier Google.JapaneseIME --version 2.29.5374.0 --urls 'https://dl.google.com/tag/s/appguid%3D%7BDDCCD2A9-025E-4142-BCEB-F467B88CF830%7D%26iid%3D%7B53E222EA-B434-7374-638B-8807C2C4DEFE%7D%26lang%3Dja%26browser%3D4%26usagestats%3D0%26appname%3DGoogle%2520%25E6%2597%25A5%25E6%259C%25AC%25E8%25AA%259E%25E5%2585%25A5%25E5%258A%259B%26needsadmin%3Dtrue%26ap%3Dexternal-stable-universal/japanese-ime/GoogleJapaneseInputSetup.exe'
```

How to encode the URL to check
---

```bash
ruby -ruri -e 'ARGV.each { |s| puts s, URI.decode_www_form_component(s), nil } ' 'https://dl.google.com/tag/s/appguid%3D%7BDDCCD2A9-025E-4142-BCEB-F467B88CF830%7D%26iid%3D%7B53E222EA-B434-7374-638B-8807C2C4DEFE%7D%26lang%3Dja%26browser%3D4%26usagestats%3D0%26appname%3DGoogle%2520%25E6%2597%25A5%25E6%259C%25AC%25E8%25AA%259E%25E5%2585%25A5%25E5%258A%259B%26needsadmin%3Dtrue%26ap%3Dexternal-stable-universal/japanese-ime/GoogleJapaneseInputSetup.exe' 'Google%20%E6%97%A5%E6%9C%AC%E8%AA%9E%E5%85%A5%E5%8A%9B'
```

```diff
-https://dl.google.com/tag/s/appguid%3D%7BDDCCD2A9-025E-4142-BCEB-F467B88CF830%7D%26iid%3D%7B53E222EA-B434-7374-638B-8807C2C4DEFE%7D%26lang%3Dja%26browser%3D4%26usagestats%3D0%26appname%3DGoogle%2520%25E6%2597%25A5%25E6%259C%25AC%25E8%25AA%259E%25E5%2585%25A5%25E5%258A%259B%26needsadmin%3Dtrue%26ap%3Dexternal-stable-universal/japanese-ime/GoogleJapaneseInputSetup.exe
+https://dl.google.com/tag/s/appguid={DDCCD2A9-025E-4142-BCEB-F467B88CF830}&iid={53E222EA-B434-7374-638B-8807C2C4DEFE}&lang=ja&browser=4&usagestats=0&appname=Google%20%E6%97%A5%E6%9C%AC%E8%AA%9E%E5%85%A5%E5%8A%9B&needsadmin=true&ap=external-stable-universal/japanese-ime/GoogleJapaneseInputSetup.exe
```

```diff
-Google%20%E6%97%A5%E6%9C%AC%E8%AA%9E%E5%85%A5%E5%8A%9B
+Google 日本語入力
```

Motivation
---
To apply https://github.com/microsoft/terminal/issues/14407#issuecomment-1995761063

May relate https://github.com/microsoft/winget-pkgs/pull/129817#issuecomment-1851012968 and https://github.com/microsoft/winget-pkgs/issues/92434

File: manifests/g/Google/JapaneseIME/2.29.5374.0/Google.JapaneseIME.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Created with komac v2.1.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: Google.JapaneseIME
+PackageVersion: 2.29.5374.0
```
Comment (kachick) [inline manifests/g/Google/JapaneseIME/2.29.5374.0/Google.JapaneseIME.installer.yaml:5]: スクリーンショット 2024-03-14 173611

Sorry, I'll fix 🙇‍♂️

File: manifests/g/Google/JapaneseIME/2.29.5370.0/Google.JapaneseIME.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Created with komac v2.1.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: Google.JapaneseIME
+PackageVersion: 2.29.5370.0
+InstallerType: exe
+ReleaseDate: 2024-01-10
+Installers:
+- Architecture: x64
+  InstallerUrl: https://dl.google.com/tag/s/appguid=%7BDDCCD2A9-025E-4142-BCEB-F467B88CF830%7D&iid=%7B53E222EA-B434-7374-638B-8807C2C4DEFE%7D&lang=ja&browser=4&usagestats=0&appname=Google%20%E6%97%A5%E6%9C%AC%E8%AA%9E%E5%85%A5%E5%8A%9B&needsadmin=true&ap=external-stable-universal/japanese-ime/GoogleJapaneseInputSetup.exe
```
[the remark this answers is not in the document]
Comment (kachick) [reply manifests/g/Google/JapaneseIME/2.29.5370.0/Google.JapaneseIME.installer.yaml:10]: iid looks like their Instance ID, bot I don't know how to get the https://dl.google.com/release2 addresses as used in Chrome manifest

[the remark this answers is not in the document]
Comment (kachick) [reply manifests/g/Google/JapaneseIME/2.29.5370.0/Google.JapaneseIME.installer.yaml]: Updated in https://github.com/microsoft/winget-pkgs/pull/144281/commits/d2b641509dd16821f328fab8fb54afd7855f5bd8

https://github.com/SpecterShell/Dumplings/blob/26a3b4f359a0fb524dc9f30dc5ae924c0f723f8a/Tasks/Google.Chrome/Script.ps1 much helped me, thank you 🙏

File: manifests/g/Google/JapaneseIME/2.29.5370.0/Google.JapaneseIME.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+InstallerType: msi
+ProductCode: '{03E6F2DF-19B2-49B8-AEFB-E47FF09635B7}'
+ReleaseDate: 2024-02-21
+AppsAndFeaturesEntries:
+- DisplayName: Google 日本語入力
+  UpgradeCode: '{C1A818AF-6EC9-49EF-ADCF-35A40475D156}'
+Installers:
+- Architecture: x64
+  InstallerUrl: https://www.google.com/dl/release2/kjspmop3m4hu2sbbaotsynsgja_2.29.5370.0/GoogleJapaneseInput64-2.29.5370.0.msi
+  InstallerSha256: 6CA10E37F4792C91DEE6D334D57AFCD3233CE520F1E42D1B2342E5508F600474
```
Comment (kachick) [inline manifests/g/Google/JapaneseIME/2.29.5370.0/Google.JapaneseIME.installer.yaml:16]: Adding https://dl.google.com/release2/ and https://edgedl.me.gvt1.com/edgedl/release2 will be better?
Old definition only have this `https://www.google.com/dl/release2`

**Comment by kachick** [comment]
>=== manifests\g\Google\JapaneseIME\2.29.5370.0 ===
Missing Properties value based on version 2.28.4650.0:
MinimumOSVersion

I have checked the returned value from Google API

```xml
<os platform="win" version="11" arch="x64" />
```

this and version=10 returns same URLs and filename, but specified version=8 and  version=7 returns `error-osnotsupported`

File: manifests/g/Google/JapaneseIME/2.29.5370.0/Google.JapaneseIME.installer.yaml (hunk 3 above)
Comment (stephengillie) [reply manifests/g/Google/JapaneseIME/2.29.5370.0/Google.JapaneseIME.installer.yaml:17]: Unfortunately, I'm also unsure of which would be better. Feel free to submit an update PR later if it should be changed.

**Comment by stephengillie** [comment]
Hi @kachick,

This manifest adds Apps and Features entries that aren't present in previous PR versions. These entries should be added to the previous versions, or removed from this version.

(Automated response - build 870.)

**Review by stephengillie — APPROVED** [review]

**Comment by kachick** [comment]
Thanks for your reviewing merging!

>This PR adds DisplayName and UpgradeCode neither of which I believe cause an issue if they're only in one PR.

2.28.4650.0 manifest uses v1.1.0 schema, it looks like providing the fields.

https://github.com/microsoft/winget-cli/blob/7cea3d9582f9bd68a4406c73f07309b1df79f120/schemas/JSON/manifests/v1.1.0/manifest.installer.1.1.0.json#L386-L427

So adding these entries in old version is also needed? I'll try 💪 

```yaml
AppsAndFeaturesEntries:
- DisplayName: Google 日本語入力
  UpgradeCode: '{C1A818AF-6EC9-49EF-ADCF-35A40475D156}'
```

**Pull request merged**